.changeset/fluffy-actors-listen.md

```diff
@@ -0,0 +1,26 @@
+---
+'@leafygreen-ui/text-input': major
+---
+
+[LG-4143](https://jira.mongodb.org/browse/LG-4143)
+
+Disabled `TextInput` component no longer renders the `disabled` attribute and instead relies on `aria-disabled`.
+
+This change is made to ensure that disabled components are still focusable to users using keyboard navigation.
+
+For more on `aria-disabled` see the [documentation on MDN](https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Attributes/aria-disabled)
+
+#### Migration guide
+
+Functionally, migration should be seamless, however there may be unit/integration/e2e tests that relied on this behavior.
+
+##### Jest/RTL
+
+Generally, only this repo should need to test that these components have a specific attribute. We recommend updating unit tests to check that some event was or was not called.
+
+However, there are cases where this may still need to be tested. In these cases, replace any `expect(textInput).toBeDisabled()` with an explicit check for `expect(textInput).toHaveAttribute("aria-disabled", "true")`.
+
+##### Cypress
```

Comment:
Love the detail here, great job

Reply:
can't claim credit since this is mostly copied from a previous PR :)